src/inventory_ui.cpp

```diff
@@ -910,7 +915,21 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
```

Comment:
Ah I see. It's effectively a no-op in this circumstance so unless there is a demonstrable performance concern it's probably best to avoid the extra complexity imposed via `found_item`.

As far as possible try and keep the two loops (map and vehicle) with the same logic as at some point it in the future it could/should probably be templated.